### DISCUSSION TITLE
Stop looking for preferred/better quality after X days of release

### DISCUSSION THREAD

**Discussion by hannonq** [pymedusa/Medusa#7469] [Ideas & Feature requests]
**Is your feature request related to a problem? Please describe.**
I currently have most of my shows with custom Allowed/Preferred qualities. Usually allowed is anything from 1080p and above, and preferred is 1080p BlueRay and all 4K options. The problem is that Medusa will try to find better qualities for every show, even the ones that I've watched ages ago. But that's not ideal for me because of storage limitations.

**Describe the solution you'd like**
I would like Medusa to provide an option of "auto archive" an episode after X days of it being released, just like the option with Proper search.  
For me, it makes sense to search for better qualities within, let's say, 14 days of an episode's release because there is a good change I haven't watched it yet.  
It would be great if it was possible to set this option per show, but a general setting like with "Proper search days" would already be a huge improvement.

**Describe alternatives you've considered**
I have been setting each episode/season/show to "Archived" manually, but that sucks because it's a never ending task. Whenever I watch an episode, I have to go to Medusa and set it to Archived.

I've also read [here](https://www.reddit.com/r/PyMedusa/comments/e8jw5h/why_is_medusa_running_a_backlog_search_when_i/fadkkd9?utm_source=share&utm_medium=web2x) that I only setting the Allowed qualities and leaving Preferred empty will prevent this behavior. But then again, that's not quite what I'm looking for. If a 4K option is available a few hours/days after the 1080p torrent comes up, I would still like to get that sweet 4K file.

**Additional context**

Branch: master  
Commit: 918cfe7ce01808499585ffe6c1356fd5ed369f76 . 
Version: 0.3.8 . 
Database: 44.14  
Python Version: | 3.7.5 (default, Oct 17 2019, 12:25:15) [GCC 8.3.0] . 
SSL Version: | OpenSSL 1.1.1d 10 Sep 2019 . 
OS: | Linux-4.15.0-58-generic-x86_64-with . 
Locale: | en_US.UTF-8 .

**Comment by Rouzax** [comment]
This could very nicely tie back in to the watched state. 😃 
https://github.com/pymedusa/Medusa/issues/3851

If episode is watched set it to archive and don't upgrade.

**Comment by hannonq** [comment]
Oh yes, that would be amazing @Rouzax :D
I don't use Trakt at the moment, but I definitely would start doing so if such a feature was available in Medusa!

**Comment by p0psicles** [comment]
Coming year I also want to release a Kodi add-on that ties into Medusa watched state
